src/Context/WatchLater.context.js

```diff
@@ -0,0 +1,52 @@
+import axios from "axios";
+import { AlertToast, SuccessToast } from "Components";
+import { createContext, useContext } from "react";
+import { useReducerContext } from "./Reducer.context";
+
+const WatchLaterContext = createContext();
+
+const WatchLaterProvider = ({ children }) => {
+  const encodedToken = localStorage.getItem("StormPlayToken");
+  const { dispatch } = useReducerContext();
+
+  const addToWatchLater = async (video) => {
+    try {
+      const response = await axios.post(
+        "/api/user/watchlater",
+        { video },
+        { headers: { authorization: encodedToken } }
```

Comment:
suggestion: add common API logic into a interceptor, and use constants file for the urls